src/dune/action_exec.ml

```diff
@@ -59,39 +40,118 @@ let exec_run ~(ectx : Context.t) ~(eenv : Context.For_exec.t) prog args =
           ]
     in
     invalid_prefix (Path.relative Path.build_dir target.name);
-    invalid_prefix (Path.relative Path.build_dir ("install/" ^ target.name)) );
+    invalid_prefix (Path.relative Path.build_dir ("install/" ^ target.name))
+
+let exec_run ~ectx ~eenv prog args =
+  validate_context_and_prog ectx.context prog;
   Process.run Strict ~dir:eenv.working_dir ~env:eenv.env
     ~stdout_to:eenv.stdout_to ~stderr_to:eenv.stderr_to
     ~stdin_from:eenv.stdin_from ~purpose:ectx.purpose prog args
 
+let exec_run_dynamic_client ~ectx ~eenv prog args =
+  validate_context_and_prog ectx.context prog;
+  let open Dune_action in
+  let to_dune_dep : Protocol.Dependency.t -> Dep.t =
+    let to_absolute_path = Stdune.Path.relative eenv.working_dir in
```

Comment:
Maybe better to avoid the word "absolute" here. "to_dune_path"? (I know, it's absolute in some sense, but it can still be confusing)